### PR TITLE
Initial Work-Server to Client Migration

### PR DESCRIPTION
**Pull request by justinszaro**
* Moves the exceptions file to the client, which now containers all the exceptions for both packages.

### CODE REVIEW

File: mirrulations-client/src/mirrclient/client.py [hunk 1]
```diff
@@ -124,7 +190,9 @@ def send_job(self, job, job_result):
             data['directory'] = self.path_generator.get_path(job_result)
 
         self._put_results(data)
-        self.put_results_to_mongo(data)
+        # self.put_results_to_mongo(data)
+        # Instead we should increment the redis counter for things downloaded
+
         comment_has_attachment = self.does_comment_have_attachment(job_result)
```
Comment (jack11wagner) [inline mirrulations-client/src/mirrclient/client.py:196]: Everything from line 196 to 204 might be better placed in its own function maybe `put_results`

File: mirrulations-client/src/mirrclient/client.py [hunk 2]
```diff
@@ -129,6 +129,38 @@ def get_job(self):
...
+        comment_has_attachment = self.does_comment_have_attachment(job_result)
+        json_has_file_format = self._document_has_file_formats(job_result)
+
+        if data["job_type"] == "comments" and comment_has_attachment:
+            self.download_all_attachments_from_comment(data, job_result)
+        if data["job_type"] == "documents" and json_has_file_format:
+            document_htm = self._get_document_htm(job_result)
+            if document_htm is not None:
+                self.download_htm(job_result)
+
```
Comment (jack11wagner) [inline mirrulations-client/src/mirrclient/client.py:162]: Wondering if we could move this functionality into a different function maybe `_put_results`

File: mirrulations-client/src/mirrclient/client.py [hunk 3]
```diff
@@ -129,6 +129,38 @@ def get_job(self):
...
+        # TODO new function for handling errors
+        # This function is not complete yet, but should replace send_job
+        data = {
+            'job_type': job['job_type'],
+            'job_id': job['job_id'],
+            'results': job_result,
+            'reg_id': job['reg_id'],
+            'agency': job['agency']
+        }
+        print(f'Sending Job {job["job_id"]} to Work Server')
```
Comment (jack11wagner) [inline mirrulations-client/src/mirrclient/client.py:142]: Update this print statement.

File: mirrulations-client/src/mirrclient/client.py [hunk 4]
```diff
@@ -129,6 +129,38 @@ def get_job(self):
...
+        if 'error' in job_result:
+            # Handles errors in job_results
+            result = self.redis.hget('jobs_in_progress', job.get('job_id'))
```
Comment (jack11wagner) [inline mirrulations-client/src/mirrclient/client.py:145]: This code is saying if an error is found, we want to get the url from the `jobs_in_progress` redis hash with `job_id` key and set the `invalid_jobs` redis hash key `job_id` equal to the job_id's corresponding url.

Basically saying `invalid_jobs['job_id'] `= `jobs_in_progress['job_id'] `
Where `invalid_jobs` and `jobs_in_progress` are redis hashes.

**Comment by nikovacs** [comment]
Mongo no go. Merging

**Pull request merged**